commit/merkleroot/observation.go

```diff
@@ -55,6 +59,52 @@ func (w *Processor) Observation(
 	return observation, nil
 }
 
+// initializeRMNController initializes the RMN controller iff:
+// 1. RMN is enabled.
+// 2. RMN controller is not already initialized.
+// 3. RMN remote config is available from previous outcome.
+func (w *Processor) initializeRMNController(ctx context.Context, prevOutcome Outcome) error {
+	if !w.offchainCfg.RMNEnabled {
+		return nil
+	}
+
+	if w.rmnControllerInitialized {
+		return nil
+	}
+
+	if prevOutcome.RMNRemoteCfg.IsEmpty() {
+		w.lggr.Debugw("RMN remote config is empty, skipping RMN controller initialization in this round")
+		return nil
+	}
+
+	w.lggr.Infow("Initializing RMN controller", "rmnRemoteCfg", prevOutcome.RMNRemoteCfg)
+
+	rmnNodesInfo, err := w.rmnHomeReader.GetRMNNodesInfo(prevOutcome.RMNRemoteCfg.ConfigDigest)
+	if err != nil {
+		return fmt.Errorf("failed to get RMN nodes info: %w", err)
+	}
+
+	peerIDs := make([]string, 0, len(rmnNodesInfo))
+	for _, node := range rmnNodesInfo {
+		peerIDs = append(peerIDs, node.PeerID.String())
+	}
+	for _, p2pID := range w.oracleIDToP2pID {
+		peerIDs = append(peerIDs, p2pID.String())
+	}
```

Comment:
Why do we need the P2P ID's of the CCIP oracles? AFAIK we don't communicate with them using these RMN-specific streams.

Reply:
OK, I think I'm starting to remember this, lets put this comment somewhere in the interface.